Task Management Exam/Services/UserService.cs

```
@@ -0,0 +1,84 @@
+﻿
+using Microsoft.EntityFrameworkCore;
+using Task_Management_Exam.Models;
+using Task_Management_Exam.Data;
+using Microsoft.Extensions.Configuration;
+using UserRole = Task_Management_Exam.Models.UserRole;
+
+
+namespace Task_Management_Exam.Services;
+
+public class UserService
+{
+    private readonly AppDbContext db;
+    private User? currentUser;
+
+    public UserService(AppDbContext context)
+    {
+        db = context;
+        db.Database.EnsureCreated(); 
+    }
+
+    public bool Register(string username, string password, UserRole role = UserRole.Regular)
+    {
+        if (db.Users.Any(u => u.Username == username))
+        {
+            Console.WriteLine("User already exists.");
+            return false;
```

You should throw exception instead of returning false